### ISSUE TITLE
check_rounds should refuse to run if there are unfinished persons

### ISSUE THREAD

**Issue by jfly** [thewca/worldcubeassociation.org#868]
Right now, `check_rounds` assumes that everybody has a `personId` assigned, which can cause confusing errors if that's not the case, confusing new Results team members.

@jfly:

> Here's my understanding of what happened (I can't understand code without links to the actual code =)). The error Baiqiang saw occurs when we're looking at a first round that is not a qualification round where $nbTotalPersons does not equal $nbPersons.
> 
> $nbPersons comes from this query which simply counts the number of results in a given round.
> 
> $nbTotalPersons comes from this query which counts the number of distinct result.personId in all rounds of a given event.
> 
> Consider a simple case: A competition with exactly one round of 3x3 with 2 competitors, both of whom do not have a WCA ID. Then $nbPersons = 2, and $nbTotalPersons = 1, and you'd get this error about missing competitors.
> 
> What exactly do you want this script to do instead? It doesn't really feel to me like there's a problem with the check_rounds script. I think we have a fundamental issue with our scripts which is that they're broken up and it's possible to run them out of order.

@SAuroux:

> yes, you are basically correct. These errors are not designed for the case which triggered them here for Baiqiang. 
> 
> When porting the scripts to rails, we should make sure that we really assigned all newcomers, before we can proceed, or at least give a warning in case that there are still entries in the Results table without personId.

@SAuroux, are there other pairs of scripts for which we should add guards like this (make sure A has finished running before running B)?

Unfortunately, adding checks means we have to write _more_ code. One possible implementation that would ultimately mean writing _less_ code would be to combine scripts (we'd need to be careful to make sure that this doesn't make things take too long).

**Comment by SAuroux** [comment]
Mainly the newcomer script has to be finished before proceeding, the rest
is quite exchangable.

An alternative idea would be to simply display in the newcomer script
something like "Displaying the first 20 out of 23 remaining newcomers."

2016-09-07 19:35 GMT+02:00 Jeremy Fleischman <notifications@github.com>:

**Comment by jfly** [comment]
Closing this as a duplicate of #80
